imports/client/components/AccountForm.tsx

```diff
@@ -1,20 +1,50 @@
 import { Meteor } from 'meteor/meteor';
 import { Accounts } from 'meteor/accounts-base';
-import PropTypes from 'prop-types';
-import React from 'react';
-import Alert from 'react-bootstrap/lib/Alert';
+import * as PropTypes from 'prop-types';
+import * as React from 'react';
+import * as Alert from 'react-bootstrap/lib/Alert';
 
 /* eslint-disable max-len, jsx-a11y/href-no-hash, jsx-a11y/anchor-is-valid, jsx-a11y/label-has-associated-control, jsx-a11y/label-has-for */
 
-class AccountForm extends React.Component {
+enum AccountFormFormat {
+  LOGIN = 'login',
+  REQUEST_PW_RESET = 'requestPwReset',
+  ENROLL = 'enroll',
+  RESET_PWD = 'resetPwd',
+}
+
+interface AccountFormProps {
+  format: AccountFormFormat;
+  onFormatChange: Function;
+  token: string;
+}
+
+enum AccountFormSubmitState {
+  IDLE = 'idle',
+  SUBMITTING = 'submitting',
+  FAILED = 'failed',
+  SUCCESS = 'success',
+}
+
+interface AccountFormState {
+  submitState: AccountFormSubmitState;
+  errorMessage: string;
+  successMessage: string;
+  email: string;
+  password: string;
+  displayName: string;
+  phoneNumber: string;
+}
+
+class AccountForm extends React.Component<AccountFormProps, AccountFormState> {
   static propTypes = {
```

Comment:
I'm not sure what the right strategy is for proptypes with TS. Arguably if the whole codebase was converted we could just drop them, I guess?